### ISSUE TITLE
why some classes aren't in the default nuget packages?

### ISSUE THREAD

**Issue by luisrudge** [Azure/servicefabric-samples#1]
Hi, classes like `ServiceEventSource.cs` and `OwinCommunicationListener.cs` are being used in many of the samples projects and I believe they'll be used in many many many services out there. Maybe they should be inside the nuget packages?

**Comment by roja** [comment]
Totally agree.

Same with ActorEventSource

**Comment by vturecek** [comment]
ServiceEventSource.cs is actually there for you to extend if you want to follow the EventSource pattern for ETW https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventsource(v=vs.110).aspx

Stay tuned for more information on an ASP.NET NuGet package.

**Comment by varghesep** [comment]
A Service Fabric OWINCommunicationListener NuGet is what needed. Bringing ASP.NET NuGet may bring more baggage that depends on ASP.NET.

**Comment by StenPetrov** [comment]
Whe whole System.Fabric is missing from the nuget package, it's referenced from GAC for some reason. Wondering - what would that reason be? It's causing issues with my build setup

Looking at *.csproj: 
```xml
<Reference Include="System.Fabric" />

<Reference Include="System.Fabric.Common.Internal"> 
  <HintPath>$(SolutionDir)packages\Microsoft.ServiceFabric.4.0.1427-preview1\lib\net45\System.Fabric.Common.Internal.dll</HintPath> 
</Reference> 
```

**Comment by vturecek** [comment]
The runtime installation is in the GAC today, yes. The service and actor frameworks and their components for writing services on the runtime are distributed in NuGet packages.

**Comment by StenPetrov** [comment]
Any particular reason for this to be the case?

**Comment by masnider** [comment]
@StenPetrov largely legacy reasons. Service Fabric as an internal Microsoft project is several years old and deployed widely internally, so we're working on a migration path where things continue to work without relying on the implicit GAC behavior. That said if you ran through the normal runtime install process you should not be experiencing issues with VS being unable to find things in the normal case. Can you share the steps you are trying? @VipulM-MSFT / @seanmck fyi

**Comment by StenPetrov** [comment]
Yes, running the SDK works fine on a dev box, but definitely something I wouldn't like to manage on a build server. 
The nuget package name is a bit misleading as it doesn't contain SF itself

**Comment by reifnir** [comment]
As well as ServiceBusCommunicationListener whenever that's provided

**Comment by isaacabraham** [comment]
+1 for NuGet packages for the actual service fabric itself.
